### PR TITLE
Bad lasso

### LINKED ISSUE #6749: Lasso highlighting - bad line separation, bad arc trimming
**Describe the bug**
Some lasso highlight situations cause the lines of the lassos round the same atoms to be too close together.
Some lasso highlight situations have full circles at joins, which shouldn't happen

**To Reproduce**
```
from rdkit import Chem
from rdkit.Chem import ChemicalFeatures
from rdkit import RDConfig
from collections import defaultdict
import os
from rdkit import Chem
from rdkit.Chem import Draw
from rdkit.Chem.Draw import IPythonConsole
from rdkit.Chem import rdDepictor
from rdkit import rdBase
print(rdBase.rdkitVersion)

IPythonConsole.molSize = 300,250
vemurafenib = Chem.MolFromSmiles('CCCS(=O)(=O)Nc1ccc(F)c(c1F)C(=O)c2c[nH]c3c2cc(cn3)c4ccc(Cl)cc4')

fdefName = os.path.join(RDConfig.RDDataDir,'BaseFeatures.fdef')
factory = ChemicalFeatures.BuildFeatureFactory(fdefName)
feats = factory.GetFeaturesForMol(vemurafenib)

colors = {'Donor':(1.,.2,1),
         'Acceptor':(.2,1,1),
         'Aromatic':(.8,.8,.2),
         'LumpedHydrophobe':(.4,.4,.2)}
ats = defaultdict(list)
bonds = defaultdict(list)
arads = {}
lws = {}
for feat in feats:
    fam = feat.GetFamily()
    if fam not in colors:
        continue
    for aid in feat.GetAtomIds():
        ats[aid].append(colors[fam])
        arads[aid] = 0.2
        for aid2 in feat.GetAtomIds():
            if aid2>aid:
                bnd = vemurafenib.GetBondBetweenAtoms(aid,aid2)
                if bnd:
                    bonds[bnd.GetIdx()].append(colors[fam])

ats = dict(ats)
bonds = dict(bonds)
from rdkit.Chem.Draw import rdMolDraw2D
from IPython.display import SVG
d2d = rdMolDraw2D.MolDraw2DSVG(600,400)
d2d.drawOptions().useBWAtomPalette()
d2d.drawOptions().fillHighlights = False
d2d.drawOptions().multiColourHighlightStyle = Draw.MultiColourHighlightStyle.Lasso
d2d.DrawMoleculeWithHighlights(vemurafenib, "", ats, bonds, arads, lws)
d2d.FinishDrawing()


SVG(d2d.GetDrawingText())
```

**Expected behavior**
It should look better

**Screenshots**
image

**Configuration (please complete the following information):**
 - RDKit version:  master branch
 - OS: macOS and whatever Greg uses
 - Python version (if relevant): it's in the C++ layer
 - Are you using conda? no
 - If you are using conda, which channel did you install the rdkit from?
 - If you are not using conda: how did you install the RDKit? built from source

**Additional context**
This code doesn't produce quite this picture on my Mac, but it's what Greg found originally.

### PR DESCRIPTION
**Pull request by DavidACosgrove**
#### Reference Issue
Fixes #6749

#### What does this implement/fix? Explain your changes.
Separates the lasso lines better.
Trims arcs/circles correctly.

#### Any other comments?
Code from issue now produces this on my Mac.
image

### CODE REVIEW

**Review by greglandrum — CHANGES_REQUESTED** [review]
The output looks good, but I have some questions/comments about the implementation

File: Code/GraphMol/MolDraw2D/DrawMolMCHLasso.cpp [hunk 1]
```diff
@@ -146,23 +155,36 @@ void DrawMolMCHLasso::drawLasso(size_t lassoNum, const RDKit::DrawColour &col,
...
+}  // namespace
+
 // ****************************************************************************
 void DrawMolMCHLasso::extractAtomArcs(
     size_t lassoNum, const RDKit::DrawColour &col,
     const std::vector<int> &colAtoms,
     std::vector<std::unique_ptr<DrawShapeArc>> &arcs) const {
-  double xradius, yradius;
+  // an empirically derived lineWidth.
+  int lineWidth = 3;
```
[the remark this answers is not in the document]
Comment (DavidACosgrove) [reply Code/GraphMol/MolDraw2D/DrawMolMCHLasso.cpp:177]: It does scale in this instance, but I will make that more obvious.  The lines look sensible on 1200x800, 600x300 and 300x200 canvases.

File: Code/GraphMol/MolDraw2D/DrawMolMCHLasso.cpp [hunk 2]
```diff
@@ -146,23 +155,36 @@ void DrawMolMCHLasso::drawLasso(size_t lassoNum, const RDKit::DrawColour &col,
   }
 }
 
+namespace {
+double getLassoWidth(const DrawMolMCH *dm, int atNum, int lassoNum) {
+  double xrad, yrad;
+  dm->getAtomRadius(atNum, xrad, yrad);
+  double rats[] = {1.0, 1.414, 2, 2.828, 4};
```
[the remark this answers is not in the document]
Comment (DavidACosgrove) [reply Code/GraphMol/MolDraw2D/DrawMolMCHLasso.cpp:162]: If my maths is correct, this doubles the area of the lasso circles, which seems to make for a reasonable effect.

**Comment by greglandrum** [comment]
lasso_highlights_7.svg has the following bad line in it:
image
(the orange line below the H on N15

**Comment by greglandrum** [comment]
The drawing of the reported problem case (`bad_lasso_1.svg`) now looks quite good, IMO.

**Comment by DavidACosgrove** [comment]
> lasso_highlights_7.svg has the following bad line in it:
> image
> (the orange line below the H on N15

Groan. I thought I’d cleared all those. I’ll see what can be done, but it’ll be a few days, probably.

**Review by greglandrum — APPROVED** [review]
LGTM

**Pull request merged**